From a740094eec8a345609c708a7cbc7825f5c03e754 Mon Sep 17 00:00:00 2001
From: mytja <mitja@severkar.eu>
Date: Sun, 22 Dec 2024 13:25:07 +0100
Subject: [PATCH 2/2] Properly apply TD patches for Lineage

- Re-implement fnmatch-like behaviour for RRO java-side
- FOD support for Asus, Samsung recent Xiaomi and RedMagic 8 devices

Makes UDFPS & overlays work again.
---
 .../pm/parsing/FrameworkParsingPackageUtils.java  |  2 +-
 .../systemui/biometrics/UdfpsController.java      |  4 ++--
 .../com/android/systemui/biometrics/UdfpsView.kt  | 15 +++++++--------
 3 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/core/java/android/content/pm/parsing/FrameworkParsingPackageUtils.java b/core/java/android/content/pm/parsing/FrameworkParsingPackageUtils.java
index d59ce04d5..0162ade1f 100644
--- a/core/java/android/content/pm/parsing/FrameworkParsingPackageUtils.java
+++ b/core/java/android/content/pm/parsing/FrameworkParsingPackageUtils.java
@@ -214,7 +214,7 @@ public class FrameworkParsingPackageUtils {
         }
         for (int i = 0; i < propNames.length; i++) {
             final String value = propValues[i];
-            final String currValue = SystemProperties.get(value);
+            final String currValue = SystemProperties.get(propNames[i]);
             // 1. Make sure prop is set.
             if (currValue == null) {
                 return false;
diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsController.java b/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsController.java
index 148d5659e..903b7e1dd 100644
--- a/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsController.java
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsController.java
@@ -784,8 +784,8 @@ public class UdfpsController implements DozeReceiver, Dumpable {
             @NonNull UdfpsOverlayInteractor udfpsOverlayInteractor,
             @NonNull PowerInteractor powerInteractor,
             @Application CoroutineScope scope,
-            @NonNull ColorDisplayManager colorDisplayManager,
-            @NonNull AuthController authController) {
+            @NonNull AuthController authController,
+            @NonNull ColorDisplayManager colorDisplayManager) {
         mContext = context;
         mExecution = execution;
         mVibrator = vibrator;
diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsView.kt b/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsView.kt
index fc6b1364f..547af3852 100644
--- a/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsView.kt
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/UdfpsView.kt
@@ -232,6 +232,13 @@ class UdfpsView(
     fun configureDisplay(onDisplayConfigured: Runnable) {
         isDisplayConfigured = true
         animationViewController?.onDisplayConfiguring()
+        ghbmView?.let { view ->
+            view.setGhbmIlluminationListener(this::doIlluminate)
+            view.visibility = VISIBLE
+            view.startGhbmIllumination(onDisplayConfigured)
+        } ?: {
+            doIlluminate(null /* surface */, onDisplayConfigured)
+        }
 
         mySurfaceView.setVisibility(VISIBLE)
         Log.d("PHH", "setting surface visible!")
@@ -311,14 +318,6 @@ class UdfpsView(
             Log.d("PHH-Enroll", "Nubia scenario in UdfpsView reached!")
             File(nubiaHbmState).writeText("4095")
           }
-
-        ghbmView?.let { view ->
-            view.setGhbmIlluminationListener(this::doIlluminate)
-            view.visibility = VISIBLE
-            view.startGhbmIllumination(onDisplayConfigured)
-        } ?: {
-            doIlluminate(null /* surface */, onDisplayConfigured)
-        }
     }
 
     private fun doIlluminate(surface: Surface?, onDisplayConfigured: Runnable?) {
-- 
2.34.1

